great_expectations/data_context/types/base.py

```diff
@@ -15,88 +20,536 @@
+        )
+        datasource_config: dict = {
+            "module_name": module_name,
+            "class_name": class_name,
+            "data_asset_type": {
+                "class_name": "SparkDFDataset",
+                "module_name": "great_expectations.dataset",
+            },
+        }
+        datasource_config.update(**kwargs)
+        self.add_datasource(name=name, **datasource_config)
+
+    def add_validation_operator(
+        self, validation_operator_name: str, validation_operator_config: dict
+    ):
+        self.validation_operators[validation_operator_name] = validation_operator_config
+
+    def add_action_list_validation_operator(
+        self, name: str, slack_webhook: str = None, slack_notify_on: str = "all"
+    ):
+        logger.debug(
+            f"Starting DataContextConfig.add_action_list_validation_operator for validation_operator_name {name}"
+        )
+        action_list: list = [
+            {
+                "name": "store_validation_result",
+                "action": {
+                    "module_name": "great_expectations.validation_operators.actions",
+                    "class_name": "StoreValidationResultAction",
+                },
+            },
+            {
+                "name": "store_evaluation_params",
+                "action": {
+                    "module_name": "great_expectations.validation_operators.actions",
+                    "class_name": "StoreEvaluationParametersAction",
+                },
+            },
+            {
+                "name": "update_data_docs",
+                "action": {
+                    "module_name": "great_expectations.validation_operators.actions",
+                    "class_name": "UpdateDataDocsAction",
+                },
+            },
+        ]
+
+        # noinspection SpellCheckingInspection
+        notify_slack_action_dict: dict = {
+            "name": "notify_slack",
+            "action": {
+                "module_name": "great_expectations.validation_operators.actions",
+                "class_name": "SlackNotificationAction",
+                "slack_webhook": slack_webhook,
+                "notify_on": slack_notify_on,
+                "renderer": {
+                    "module_name": "great_expectations.render.renderer.slack_renderer",
+                    "class_name": "SlackRenderer",
+                },
+            },
+        }
+
+        if slack_webhook is not None:
+            action_list.append(notify_slack_action_dict)
+
+        validation_operator_config: dict = {
+            "module_name": "great_expectations.validation_operators.validation_operators",
+            "class_name": "ActionListValidationOperator",
+            "action_list": action_list,
+        }
+
+        self.add_validation_operator(
+            validation_operator_name=name,
+            validation_operator_config=validation_operator_config,
+        )
+
+    @property
```

Comment:
I agree with you.  I would prefer it to be the same.  Unfortunately, we have problems with circular dependencies.  How about we revisit in a subsequent development cycle and fix all of it?

Reply:
Sure, I don't mean to hold this up. Here is what I suggested for circular dependencies in another comment: What about trying something like this to avoid circular dependency issues? https://stackoverflow.com/a/39757388/2916375 i.e. shielding the imports during runtime and only importing for the type checker using a pattern like:
```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from main import Main
```

Reply:
@anthonyburdi and @alexsherstinsky discussed this: deferred (we can refactor the code in a separate development cycle).